csrc/scheduler/normalization_utils.cpp

```diff
@@ -702,7 +707,11 @@ getOptionalInnerOuterPersistentBufferBatches(
   // 320 bytes stack frame, 320 bytes spill stores, 640 bytes spill loads. As a
   // ref, the segmented version takes time_us mean(var)= 2841.91 (5.20231)
   // without considering the overhead of fusion segmentation.
-  if (ignore_register_size_limit || inner_batch <= batch_max + 3) {
+  // (4) Disable this optimization if vectorize_factor is 1 due to high register
+  // usage in cases can't be vectorized.
+  const int64_t batch_max_reg_spill =
+      vectorize_factor > 1 ? batch_max + 3 : batch_max;
+  if (ignore_register_size_limit || inner_batch <= batch_max_reg_spill) {
```

Comment:
This was observed in this case and previously when we investigated layer norm forward.

Reply:
How does the register usage look like without vectorization? How much impact of allowing extra 3 registers here does it actually mean for actual register usage?

Reply:
>How does the register usage look like without vectorization?

I compared cases with hidden size of 7119 (63x113 can't be vectorized) and 7232 (64x113 vectorized by 8).
Case-7119 stores 1 x 45 persistent batches and used 253 registers. Estimated is 45*14/4 + 40 = 198
Case-7232 stores 8 x 6   persistent batches and used 240 registers. Estimated is 48*14/4 + 40 = 208

>How much impact of allowing extra 3 registers here does it actually mean for actual register usage?

This 3 is extra 3 batches. It was used before this PR to avoid segmentation for hidden size >=14K and increased performance. However, if we use this for unvectorized cases, the performance is worse than segmented version due to sevier register spills. e.g. for case with hidden size of 14049, allowing extra 3 batches will lead to unsegmented version and the bandwidth is only 351.382G/s due to register spills `ptxas         .     152 bytes stack frame, 152 bytes spill stores, 300 bytes spill loads`. Howeve, if don't allow extra 3 batches, the segmented version can get 465.255G/s.
By comparasion, the vectorized version with hidden size of 14272, the bandwith is 1.14859T/s without spill.

Reply:
My bad. I mean 45 x 14 / 4 + 40 = 198. Somehow I used * and it changed to italic the text. 45 is total batch size, 14 is bytes per batch, 4 is bytes per register. Same for 4814/4 + 40 = 208